sdk/core/Azure.Core/api/Azure.Core.net6.0.cs

```diff
@@ -544,8 +544,11 @@ public abstract partial class RequestContent : System.IDisposable
         public static Azure.Core.RequestContent Create(byte[] bytes) { throw null; }
         public static Azure.Core.RequestContent Create(byte[] bytes, int index, int length) { throw null; }
         public static Azure.Core.RequestContent Create(System.IO.Stream stream) { throw null; }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Calls JsonSerializer.SerializeToUtf8Bytes which requires unreferenced code.")]
```

Comment:
I would recommend using a better message here. The guidance is that this should be actionable. The UX is that when the user of the library calls the method they will get a warning with this message in it inside the IDE, their build and publish. And it will point to the line of code which calls this method.

The user is generally not interested in why it doesn't work in AOT/Trimming, they may not even know what "unreferenced code" really means.

A better message would simply state that this method is not supported with trimming and what should the user do instead - call a different method perhaps, or use a different pattern. Please note that the attribute also takes a Url property which will be included in the warning message if specified. So if the solution for the user is more complicated, the message can be relatively simple and the Url can point to a more detailed description.